### PR TITLE
Fix doc CI for macOS and Windows

### PR DESCRIPTION
**Pull request by michdolan**
The issue ended up being use of deprecated pip/setup argument (in `FindSphinx.cmake`) and the Windows and macOS VMs recently bumping to the pip version (20.2) which removed the deprecated feature. On Windows there was a second issue which is that the pip command executed by CMake needed parameter paths to use the native Windows path separator (`\\` instead of `/`). With those resolved, the docs build on all platforms with the most recent pip release. The docs are still disabled for Python 2 however due to incompatibility of the `sphinx-press-theme` package. I disabled the doc build on all Python 2 CI builds to make this expected behavior. These fixes _should_ alleviate local Sphinx build issues as well. This PR also fixes the nightly "latest" build which needed the required python packages installed explicitly.

Signed-off-by: Michael Dolan <michdolan@gmail.com>

### CODE REVIEW

**Comment by hodoulp** [comment]
> These fixes should alleviate local Sphinx build issues as well.

Using `cmake` on Windows platform, the configuration step still fails to find some packages (i.e. `testresources`, `recommonmark` and `sphink_press_theme`) but the HTML documentation is correctly generated.

**Comment by hodoulp** [comment]
Question: When using `Python` version 3 the default pip name is `pip3` so the use of `pip` could be ambiguous, should the code use `pip3` (e.g. share/ci/scripts/linux/install_recommonmark.sh) ?

Perhaps the version agnostic usage could be better i.e. `python -m pip install --upgrade pip setuptools xxx yyy`.

**Comment by doug-walker** [comment]
Very helpful, thanks Michael!

**Comment by michdolan** [comment]
Good call with accessing pip via python. I will try that. @hodoulp in your instance of cmake not finding the python packages, are you using `OCIO_INSTALL_EXT_PACKAGES`, or providing your own? The cmake module attempts to import them via the python interpreter, so should detect them if they are already installed and cmake finds your python interpreter. Could you share the specific error and env details so I can troubleshoot? Thanks!

**Comment by hodoulp** [comment]
@michdolan I use `cmake -DOCIO_BUILD_DOCS=ON ..\.` without any packages already installed in Python.

On the Windows platform the setup is:
```
(pr_1090) F:\OpenSource\ocio_1\build_win_rls> python --version
Python 3.8.5

(pr_1090) F:\OpenSource\ocio_1\build_win_rls> pip --version
pip 20.2.1 from f:\python38\lib\site-packages\pip (python 3.8)

(pr_1090) F:\OpenSource\ocio_1\build_win_rls> pip list
Package    Version
---------- -------
-ip        20.1.1
pip        20.2.1
setuptools 47.1.0

(pr_1090) F:\OpenSource\ocio_1\build_win_rls>
```
The error messages are:
```
-- Installing pybind11: F:/OpenSource/ocio_1/build_win_rls/ext/dist/include (version "2.4.3")
-- Could NOT find Sphinx (missing: Sphinx_EXECUTABLE) (Required is at least version "2.0.0")
-- Installing Sphinx: F:/OpenSource/ocio_1/build_win_rls/ext/dist/Scripts/sphinx-build (version "2.0.0")
CMake Warning at share/cmake/macros/FindPythonPackage.cmake:88 (message):
  Could NOT find testresources: import testresources failed
Call Stack (most recent call first):
  share/cmake/modules/FindExtPackages.cmake:56 (find_python_package)
  CMakeLists.txt:260 (include)


-- Installing testresources: F:/OpenSource/ocio_1/build_win_rls/ext/dist/lib/site-packages (version 2.0.1)
CMake Warning at share/cmake/macros/FindPythonPackage.cmake:88 (message):
  Could NOT find recommonmark: import recommonmark failed
Call Stack (most recent call first):
  share/cmake/modules/FindExtPackages.cmake:64 (find_python_package)
  CMakeLists.txt:260 (include)


-- Installing recommonmark: F:/OpenSource/ocio_1/build_win_rls/ext/dist/lib/site-packages (version 0.6.0)
CMake Warning at share/cmake/macros/FindPythonPackage.cmake:88 (message):
  Could NOT find sphinx-press-theme: import sphinx_press_theme failed
Call Stack (most recent call first):
  share/cmake/modules/FindExtPackages.cmake:68 (find_python_package)
  CMakeLists.txt:260 (include)


-- Installing sphinx-press-theme: F:/OpenSource/ocio_1/build_win_rls/ext/dist/lib/site-packages (version 0.5.1)
-- Found Python: F:/Python38/python.exe (found version "3.8.5") found components:  Interpreter
-- Could NOT find OpenImageIO (missing: OpenImageIO_DIR)
-- Use "OpenImageIO_ROOT" to specify an install location
CMake Warning at src/apps/CMakeLists.txt:17 (message):
  Could NOT find OpenImageIO.  Skipping build for ociolutimage, ocioconvert,
  ociodisplay and ocioperf


-- Found Python: F:/Python38/python.exe (found version "3.8.5") found components:  Interpreter Development
-- Found Python: F:/Python38/python.exe (found version "3.8.5") found components:  Interpreter
-- Create sphinx conf.py from conf.py.in
```

**Comment by hodoulp** [comment]
Concerning the ubuntu platform, upgrading the  packages (i.e. `python -m pip install --upgrade pip testresources recommonmark sphinx_press_theme`) outside and then running `cmake -DOCIO_BUILD_DOCS=ON ../.` perfectly works.
```
root@86e4df61daaa:/home/devel/ocio_1/build# pip list
Package                       Version
----------------------------- ---------
alabaster                     0.7.12
Babel                         2.8.0
certifi                       2020.6.20
chardet                       3.0.4
commonmark                    0.9.1
docutils                      0.16
idna                          2.10
imagesize                     1.2.0
Jinja2                        2.11.2
MarkupSafe                    1.1.1
packaging                     20.4
pbr                           5.4.5
pip                           20.2.1
Pygments                      2.6.1
pyparsing                     2.4.7
pytz                          2020.1
recommonmark                  0.6.0
requests                      2.24.0
setuptools                    45.2.0
six                           1.15.0
snowballstemmer               2.0.0
Sphinx                        3.2.0
sphinx-press-theme            0.5.1
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        1.0.3
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.4
testresources                 2.0.1
urllib3                       1.25.10
wheel                         0.34.2
root@86e4df61daaa:/home/devel/ocio_1/build# 
```

**Comment by michdolan** [comment]
The cmake configure messages you posted above are warnings, so should not cause the configure step to fail. For example, the message below is logging that the CMake `FindPyhtonPackage` module tried to import ``testresources`` (to see if it is installed), failed, and then triggered ``testresources`` to install at build time (via pip). That's the effect of the default ``-DOCIO_INSTALL_EXT_PACKAGES=MISSING`` option. 
```
CMake Warning at share/cmake/macros/FindPythonPackage.cmake:88 (message):
  Could NOT find testresources: import testresources failed
Call Stack (most recent call first):
  share/cmake/modules/FindExtPackages.cmake:56 (find_python_package)
  CMakeLists.txt:260 (include)


-- Installing testresources: F:/OpenSource/ocio_1/build_win_rls/ext/dist/lib/site-packages (version 2.0.1)
```
Those warnings don't show up in our CI because we set ``-DOCIO_INSTALL_EXT_PACKAGES=ALL``, so that an import is never attempted at configure time. The package is always then installed at build time. If you trigger the OCIO build following the CMake configure containing those warnings, it should install them for you and succeed. If not though, I'd be curious to see the error.

So basically, whenever you see ``-- Installing ...`` following a CMake warning, that means it will be installed for you. If you want to always install them, and not have any check warnings, specify ``-DOCIO_INSTALL_EXT_PACKAGES=ALL``.

If this is confusing, we could silence the warnings. I hadn't done so before to try and communicate when we will be using an installed package vs a system one. Thanks for all the details @hodoulp .

As for this PR, I think we should go ahead and merge it to unblock the CI, and we can refine the build and messaging further in a later PR.

**Comment by hodoulp** [comment]
Thanks for the explanations @michdolan. The messages are quite confusing but the improvement could be done later to be be ready for DigiPro.
